fmriprep/workflows/fieldmap/phdiff.py

```diff
@@ -110,16 +102,44 @@ def _pick1st(inlist):
     # pre_fugue = pe.Node(fsl.FUGUE(save_fmap=True), name='ComputeFieldmapFUGUE')
     # rsec2hz (divide by 2pi)
 
+    if phasetype == "phasediff":
+        # phase diff -> radians
+        pha2rads = pe.Node(niu.Function(function=siemens2rads),
+                           name='pha2rads')
+        # Read phasediff echo times
+        meta = pe.Node(ReadSidecarJSON(), name='meta', mem_gb=0.01,
+                       run_without_submitting=True)
+        workflow.connect([
+            (meta, compfmap, [('out_dict', 'metadata')]),
+            (inputnode, pha2rads, [('phasediff', 'in_file')]),
+            (pha2rads, prelude, [('out', 'phase_file')]),
+            (inputnode, ds_fmap_mask, [('phasediff', 'source_file')]),
+        ])
+
+    elif phasetype == "phase":
+        workflow.__desc__ += """\
+The phase difference used for unwarping was calculated using a method developed
+by Mark Elliott [@pncprocessing].
+    """
+        # Special case for phase1, phase2 images
+        meta = pe.MapNode(ReadSidecarJSON(), name='meta', mem_gb=0.01,
+                          run_without_submitting=True, iterfield=['in_file'])
+        phases2fmap = pe.Node(Phases2Fieldmap(), name='phases2fmap')
+        workflow.connect([
+            (meta, phases2fmap, [('out_dict', 'metadatas')]),
+            (inputnode, phases2fmap, [('phasediff', 'phase_files')]),
+            (phases2fmap, prelude, [('out_file', 'phase_file')]),
```

Comment:
the difference map needs to be phase unwrapped? aren't the phase maps unwrapped individually?

Reply:
Consider:

p1 = pi - eps
p2 = -pi + eps

p1 - p2 = 2pi - 2eps = -2eps

Phase differences should be between -pi and pi, as well.

Reply:
That means you are clipping it again, correct?

Reply:
Oh sorry, I think I misunderstood the context here. You're saying that `prelude` just does the clipping between -pi and pi, and we've already done that in `phases2fmap`?

Reply:
Yes, I understood before that each phase map needs to be unwrapped (which makes sense to me). If they are phase-unwrapped, there's no point on running [PRELUDE](https://fsl.fmrib.ox.ac.uk/fsl/fslwiki/FUGUE/Guide#PRELUDE_.28phase_unwrapping.29) (Phase Region Expanding Labeller for Unwrapping Discrete Estimates).

Also, I am very curious as to how the individual phase maps are unwrapped, if not done with PRELUDE. I know there are better methods out there (e.g., using graph cuts) but they are rarely used in general.

Reply:
Okay, I have confirmed this point: as per [the FUGUE guide](https://fsl.fmrib.ox.ac.uk/fsl/fslwiki/FUGUE/Guide), each phase map should be unwrapped with PRELUDE separately (step 3) of the guide. Then they can be subtracted as they are not wrapped within [-pi, pi) anymore.

After that point, the pipeline should be exactly the same as for the `phasediff` type of fieldmap.